filters/glitch/src/GlitchFilter.js

```diff
@@ -0,0 +1,421 @@
+        }, options);
+
+        this.direction = options.direction;
+        this.red = options.red;
+        this.green = options.green;
+        this.blue = options.blue;
+
+        /**
+         * The maximum offset value for each of the slices.
+         *
+         * @member {number}
+         */
+        this.offset = options.offset;
+
+        /**
+         * The fill mode of the space after the offset.
+         *
+         * @member {number}
+         */
+        this.fillMode = options.fillMode;
+
+        /**
+         * average
+         *
+         * @member {boolean}
+         */
+        this.average = options.average;
+
+        /**
+         * A seed value for randomizing color offset. Animating
+         * this value to `Math.random()` produces a twitching effect.
+         *
+         * @member {number}
+         */
+        this.seed = options.seed;
+
+        /**
+         * Minimum size of slices as a portion of the `sampleSize`
+         *
+         * @member {number}
+         */
+        this.minSize = options.minSize;
+
+        /**
+         * Height of the displacement map canvas.
+         *
+         * @member {number}
+         * @readonly
+         */
+        this.sampleSize = options.sampleSize;
+
+        /**
+         * Internally generated canvas.
+         *
+         * @member {HTMLCanvasElement} _canvas
+         * @private
+         */
+        this._canvas = document.createElement('canvas');
+        this._canvas.width = 8;
+        this._canvas.height = this.sampleSize;
+
+        /**
+         * The displacement map is used to generate the bands.
+         * If using your own texture, `slices` will be ignored.
+         *
+         * @member {PIXI.Texture}
+         * @readonly
+         */
+        this.texture = PIXI.Texture.fromCanvas(this._canvas, PIXI.SCALE_MODES.NEAREST);
+
+        /**
+         * Internal number of slices
+         * @member {number}
+         * @private
+         */
+        this._slices = 0;
+
+        // Set slices
+        this.slices = options.slices;
+    }
+
+    /**
+     * Override existing apply method in PIXI.Filter
+     * @private
+     */
+    apply(filterManager, input, output, clear) {
+
+        const width = input.sourceFrame.width;
+        const height = input.sourceFrame.height;
+
+        this.uniforms.dimensions[0] = width;
+        this.uniforms.dimensions[1] = height;
+        this.uniforms.aspect = height / width;
+
+        this.uniforms.seed = this.seed;
+        this.uniforms.offset = this.offset;
+        this.uniforms.fillMode = this.fillMode;
+
+        filterManager.applyFilter(this, input, output, clear);
+    }
+
+    /**
+     * Randomize the slices size (heights).
+     *
+     * @private
+     */
+    _randomizeSizes() {
+        const arr = this._sizes;
+        const last = this._slices - 1;
+        const size = this.sampleSize;
+        const min = Math.min(this.minSize / size, 0.9 / this._slices);
+
+        if (this.average) {
```

Comment:
@finscn, I don't understand the benefit of using `average` just different ways to slice up the total size. I'd recommend just picking one. The end result looks fine either way.

Reply:
average  will make the widths of slices is almost in same width.
Just almost.  e.g.
if average == false ,  the width maybe  from  1% to 70% .
if average == true ,  the width maybe from  20% to 30%.